### ISSUE TITLE
Using of pre-trained models to decode TEDLium test set

### ISSUE THREAD

**Issue by AlexandderGorodetski** [facebookresearch/wav2letter#671]
Hello,

I would like to use one of your pre-trained models to perform decoding of TEDLium test set. 

Could you please guide me how to do it.

I was able to perform decoding of Lirbri-Clean test set and I got an expected WER and LER.

But for Libri-Clean I have ready lexicon file (I am using word-piece model). I guess that lexicon of TED is different than lexicon of Libri-Clean. Do you have some tool for lexicon creation given pre-trained acoustic model?

Thanks a lot,
Alexander.

**Comment by tlikhomanenko** [comment]
@AlexandderGorodetski 

> Could you please guide me how to do it.

To decode model on new data you need to prepare list file as you did for librispeech where you store `id file/path duration transcription` (for example see librispeech data preparation, but my guess for TEDLium it will differ), lexicon file (see below comment on this) and possibly language model trained on your dataset lm corpus (librispeech LM possibly will work worse on out of domain data) (here you can have a look at examples how to train kenlm, or at some of our paper recipe for librispeech https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/lexicon_free/librispeech)

In the data preparation step here https://github.com/facebookresearch/wav2letter/tree/master/recipes/models/sota/2019 you can see that word-piece model (mapping of words into word-pieces) will be stored too `librispeech-train-all-unigram-10000.model`: so you need to repeat the data preparation step and have this model on your machine. Then to prepare a lexicon for new dataset you need to take all unique words from the train transcription (from test it is not fair) and/or from the language model data (if they exist like for Librispeech we have separate lm data corpus) and map them with `librispeech-train-all-unigram-10000.model` to the sequence of word-pieces. Here is part of the code which you can reuse to apply word-piece model to any words https://github.com/facebookresearch/wav2letter/blob/master/recipes/models/utilities/prepare_librispeech_wp_and_official_lexicon.py#L140 (logic is the same as we do for Libripseech, you need to switch to another set of words and take nbest=1).
